tests/integration_tests/functional/test_snapshot_version.py

```diff
@@ -0,0 +1,122 @@
+# Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+"""Basic tests scenarios for snapshot save/restore."""
+
+import logging
+import platform
+import pytest
+from conftest import _test_images_s3_bucket
+from framework.artifacts import ArtifactCollection
+from framework.builder import MicrovmBuilder, SnapshotBuilder, SnapshotType
+from framework.microvms import C3micro
+import host_tools.network as net_tools  # pylint: disable=import-error
+
+
+@pytest.mark.skipif(
+    platform.machine() != "x86_64",
+    reason="Not supported yet."
+)
+def test_restore_from_past_versions(bin_cloner_path):
```

Comment:
For the moment we don't have cross Intel families snapshot restore. So this test may fail depending on the processor model. For example on my Intel laptop it fails. Also creating snapshot artificats seems unnecessary for the moment. In order to prevent machine-dependent fails and unnecessary complexity, wouldn't it be better to just create a snapshot using the old firecracker versions (that are present as artifacts in S3) and restore it using the current version ? This would also make it easier to test cross-version snapshot restore on AMD.

Reply:
We have CPU templates so this should work, but you ar right the m5d.metal snapshots do not restore on local dev machines. I think it is a good idea to create snaps instead of getting them from S3. I will look into this.

Reply:
I implemented this as a simpler way to make this work for now.